### PR TITLE
Forward merge 0.19 into 0.20

### PR DESCRIPTION
**Pull request by raydouglass**
DO NOT USE auto-merger, this must be merged with a merge commit!

This follows up on #371

### CODE REVIEW

**Comment by raydouglass** [comment]
rerun tests

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
```
10:59:21 fatal: invalid reference: branch-.
10:59:21 CMake Error at cudf-subbuild/cudf-populate-prefix/tmp/cudf-populate-gitclone.cmake:40 (message):
10:59:21   Failed to checkout tag: 'branch-.'
```

File: cpp/cmake/thirdparty/CUSPATIAL_GetCUDF.cmake [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    cuspatial_restore_if_enabled(BUILD_TESTS)
+    cuspatial_restore_if_enabled(BUILD_BENCHMARKS)
+
+    # Make sure consumers of cuspatial can see cudf::cudf
+    fix_cmake_global_defaults(cudf::cudf)
+    # Make sure consumers of cuspatial can see cudf::cudftestutil
+    fix_cmake_global_defaults(cudf::cudftestutil)
+endfunction()
+
+set(CUSPATIAL_MIN_VERSION_cudf "${CUSPATIAL_VERSION_MAJOR}.${CUSPATIAL_VERSION_MINOR}")
```
Comment (kkraus14) [inline cpp/cmake/thirdparty/CUSPATIAL_GetCUDF.cmake:43]: I think this doesn't match the CMake project name here: https://github.com/rapidsai/cuspatial/pull/374/files#diff-1bba462ab050e89360fd88110a689e85ee037749cea091a1848ab574381d3795R31 which causes them to be empty strings

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -14,19 +14,57 @@
...
+
+# This needs to be run before enabling the CUDA language due to the default initialization behavior
+# of `CMAKE_CUDA_ARCHITECTURES`, https://gitlab.kitware.com/cmake/cmake/-/issues/21302
+if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES OR CMAKE_CUDA_ARCHITECTURES STREQUAL "ALL")
+  set(CUSPATIAL_BUILD_FOR_ALL_ARCHS TRUE)
+elseif(CMAKE_CUDA_ARCHITECTURES STREQUAL "")
+  set(CUSPATIAL_BUILD_FOR_DETECTED_ARCHS TRUE)
+endif()
 
 project(CUDA_SPATIAL VERSION 0.20.0 LANGUAGES C CXX CUDA)
```
Comment (kkraus14) [inline cpp/CMakeLists.txt:31]: I think this is a bad merge as it's renamed in 0.19: https://github.com/rapidsai/cuspatial/blob/branch-0.19/cpp/CMakeLists.txt#L31

Comment (raydouglass) [reply cpp/CMakeLists.txt]: Yeah, I guess I picked the wrong one.

**Review by trxcllnt — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**